### PR TITLE
Negate method invocation

### PR DESCRIPTION
**Pull request by skrzypo987**

### CODE REVIEW

**Comment by findepi** [comment]
This looks like a correctness fix. Is it testable?

**Comment by skrzypo987** [comment]
The actual implementation that is used is bytecode-generated. This is not used anywhere, so no tests so far

**Review by sopel39 — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/operator/SimplePagesHashStrategy.java [hunk 1]
```diff
@@ -215,7 +215,7 @@ public boolean positionNotDistinctFromRow(int leftBlockIndex, int leftPosition,
             Block leftBlock = channels.get(hashChannel).get(leftBlockIndex);
             Block rightBlock = page.getBlock(rightChannels[i]);
             BlockPositionIsDistinctFrom isDistinctFromOperator = isDistinctFromOperators.get(i);
-            if (!isDistinctFromOperator.isDistinctFrom(leftBlock, leftPosition, rightBlock, rightPosition)) {
+            if (isDistinctFromOperator.isDistinctFrom(leftBlock, leftPosition, rightBlock, rightPosition)) {
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/SimplePagesHashStrategy.java:218]: please add a test to `io.trino.operator.TestSimplePagesHashStrategy`
It's not dead code yet (although it should be)
```
        // if compilation fails, use interpreter
        return new SimplePagesHashStrategy(
```

I think we should remove that fallback

**Comment by mosabua** [comment]
👋 @skrzypo987 and @sopel39 .. can this be merged?

**Comment by sopel39** [comment]
no release notes needed

**Comment by mosabua** [comment]
Sweet.. thank @sopel39

**Pull request merged**